app/services/partners/request_create_service.rb

```diff
@@ -54,16 +54,27 @@ def populate_item_request(partner_request)
         attrs['item_id'].blank? && attrs['quantity'].blank?
       end
 
-      item_requests = formatted_line_items.map do |ira|
-        Partners::ItemRequest.new(
-          item_id: ira['item_id'],
-          quantity: ira['quantity'],
-          children: ira['children'] || [], # will create ChildItemRequests if there are any
-          name: fetch_organization_item_name(ira['item_id']),
-          partner_key: fetch_organization_partner_key(ira['item_id'])
-        )
+      items = {}
+
+      formatted_line_items.each do |input_item|
+        pre_existing_entry = items[input_item['item_id']]
+        if pre_existing_entry
+          pre_existing_entry.quantity = (pre_existing_entry.quantity.to_i + input_item['quantity'].to_i).to_s
+          # TODO: Is the following the correct way to merge?
+          pre_existing_entry.children += input_item['children'] || []
```

Comment:
We aren't sure how this child field is meant to be used, so we weren't sure whether this merge method makes sense.

Reply:
I _think_ you don't need to do this. From what I can tell, each child has a field `item_needed_diaperid` (which seems to indicate that a child can only really be associated with a single item). So when the family request is created, the logic collects all the children in the family relevant to the given item ID and passes them in. In this case, since the item ID is the same, the list children should also be the same. Probably worth some footling around to verify.

Reply:
Also probably should have tests around this.

Reply:
We chatted with Brock. It generally shouldn't happen because today the family requests controller eagerly groups the requested items (but we obvs don't want to rely on that controller logic for the model to be correctly instantiated). We think the 'correct-ish' way to merge in children if it _somehow_ occurred is to do it as a set-style operation. If the upstream code has mucked about with quantity and done it poorly, we won't be able to fix things here for it without causing more issues.

Might be a way in the future to pull more of that instantiation logic out of the controller, but we added a test for now to verify the behavior we've implemented.